src/runtime/TemporalObject.cpp

```diff
@@ -68,228 +68,270 @@ Value TemporalObject::toISOTime(ExecutionState& state, DateObject& d)
     return Value(new ASCIIString(result.data(), result.length()));
 }
 
-std::map<std::string, std::string> TemporalObject::parseValidIso8601String(ExecutionState& state, const Value& str)
-{
-    ASSERT(str.isString());
-    size_t pos = str.asString()->toNonGCUTF8StringData().find('[');
-    std::string isoDate = str.asString()->toNonGCUTF8StringData().substr(0, pos);
-    std::string extra;
 
-    if (pos != std::string::npos) {
-        extra = str.asString()->toNonGCUTF8StringData().substr(pos);
+std::string TemporalObject::getNNumberFromString(std::string& isoString, const int n, unsigned int& index)
+{
+    std::string retVal;
+    if (!TemporalObject::is_number(isoString.substr(index, n))) {
+        throw std::exception();
     }
+    retVal = isoString.substr(index, n);
+    index += n;
+    return retVal;
+}
 
-    pos = isoDate.find('T');
-
-    if (pos != std::string::npos) {
-        std::replace(isoDate.begin(), isoDate.end(), 'T', ' ');
+std::map<std::string, int> TemporalObject::getSeconds(std::string& isoString, unsigned int& index)
+{
+    int counter = 1;
+    while (TemporalObject::is_number(isoString.substr(counter, 1))) {
```

Comment:
Is there a single character chekc for digits?

Reply:
Yes `std::isdigit()`